modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/source/SourceConnectionManager.java

```diff
@@ -28,8 +30,9 @@
  */
 public class SourceConnectionManager {
 
+  private static final ReentrantLock lock = new ReentrantLock();
   private final ConnectionManager connectionManager;
-  private final Map<Reference<Object>, ConnectionHandler<Object>> connections = new ConcurrentHashMap<>();
+  private final Map<Reference<Object>, Pair<Integer, ConnectionHandler<Object>>> connections = new ConcurrentHashMap<>();
```

Comment:
is there any access outside of the `lock` to justify keeping this `ConcurrentHashMap`?